### ISSUE TITLE
vscode-eslint: Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser

### ISSUE THREAD

**Issue by Vishesh30** [typescript-eslint/typescript-eslint#2122]
What code were you trying to parse?

import { Af } from "@common/models";

What did you expect to happen?

no errors

What actually happened?
vscode-eslint works fine. Running npm script for lint is reporting issues. 

Below are 3 issues which i am facing:-
1. **Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.The file does not match your project config: src/models/BankAccount.ts.The file must be included in at least one of the projects provided. The models/* folder is excluded in tsconfig list as well as the eslintignore file , still i am getting the error. when i run the script to lint.**

2.  **Another issue which is see if i open the file i get  "Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option in your 'tsconfig' or 'jsconfig' to remove this warning." I have already set the configuration in tsconfig file but still i get this error.**

We have a Monorepo using "lerna" Our settings.json file has this config to support monorepo:-

`"eslint.workingDirectories": [{ "pattern": "./packages/*/" }],`

My configuration in .eslintrc.js file:-

```
module.exports = {
    env: {
        browser: true,
        es6: true,
        node: true
    },
    extends: [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "prettier",
        "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
        "plugin:prettier/recommended" // Enables eslint-plugin-prettier and eslint-config-prettier.
    ],
    parser: "@typescript-eslint/parser", // Specifies the ESLint parser
    parserOptions: {
        ecmaVersion: 2018, // Allows for the parsing of modern ECMAScript features
        project: "tsconfig.json",
        sourceType: "module"
    },
    plugins: ["@typescript-eslint", "prettier"],
    rules: {},
    settings: {}
};
```
My tsconfig file:-

```
{
    "compileOnSave": true,
    "compilerOptions": {
        "moduleResolution": "Node",
        "esModuleInterop": true,
        "module": "commonjs",
        "target": "es6",
        "sourceMap": true,
        "alwaysStrict": true,
        "outDir": "./build",
        "rootDir": "./src",
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "declaration": true,
        "resolveJsonModule": true,
        "incremental": true,
        "tsBuildInfoFile": "./.tsBuildInfoFile"
    },
    "files": [
        "./src/common/models/DataTypes.ts"
    ],
    "include": ["./src/**/*", "tests/InternalRequest"],
    "exclude": ["./node_modules/", "./src/tests/**/*.gen.ts", "./src/**/models/"]
}
```
Below is Lint Script which i am using:
`"lint": "eslint . --ext .ts,.tsx -o eslint_report.json -f json --quiet; exit 0",`

I have also added .eslintignore file to ignore model which is as below:-

```
# don't ever lint node_modules
node_modules
# don't lint build output
outdir
# don't lint nyc coverage output
coverage
# dont lint models
src/**/models/
```

package | version
-- | --
@typescript-eslint/parser | ^2.20.2
@typescript-eslint/eslint-plugin | ^2.20.0
TypeScript | 3.7.5
ESLint | ^6.8.0
node | ^10.15.3
npm | 6.10.1

Thanks,
Vishesh.

**Comment by Vishesh30** [comment]
@bradzacher Sorry for reposting the issue but , this time problem is happening when i run the eslint command

**Comment by bradzacher** [comment]
Your tsconfig has a glob which `exclude`s all files in the `src/models` folder, hence you get the error.

Our FAQ does a good job of explaining how to rectify this:
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md#i-get-errors-telling-me-the-file-must-be-included-in-at-least-one-of-the-projects-provided

LMK if there's anything that's unclear.

**Comment by Vishesh30** [comment]
> Your tsconfig has a glob which `exclude`s all files in the `src/models` folder, hence you get the error.
> 
> Our FAQ does a good job of explaining how to rectify this:
> https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md#i-get-errors-telling-me-the-file-must-be-included-in-at-least-one-of-the-projects-provided
> 
> LMK if there's anything that's unclear.
> 
> Eslint has the `--print-config` flag which is useful to help debug the resolved config for a file.

I read through it and i have included it in the .eslintignore file but still eslint is complaining about the error.

**Comment by Vishesh30** [comment]
@bradzacher I read through it and i have included it in the .eslintignore file but still eslint is complaining about the error.

**Comment by bradzacher** [comment]
Other suggestions for help debugging this:

The `--print-config` flag is good for interrogating the config eslint is using for any file

The `--debug` flag will also give you some deeper information into what eslint is doing exactly, like what files it is ignoring.

Our tooling is controlled and coordinated by ESLint, so it means that this is likely an issue with your configuration on that side.

**Comment by Vishesh30** [comment]
@bradzacher  ok I will try to debug and check , Any idea on 2nd issue.?
Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option in your 'tsconfig' or 'jsconfig' to remove this warning." I have already set the configuration in tsconfig file but still i get this error.

**Comment by bradzacher** [comment]
You've excluded your `src/model` files from your tsconfig, so they don't have the `experimentalDecorators` option turned on.
